dapp/src/app/(protected)/my-names/page.tsx

```diff
@@ -3,163 +3,160 @@
-        },
-    ];
+    const [selectedFilter, setSelectedFilter] = useState<GroupedNamesFilter>(
+        GroupedNamesFilter.All,
+    );
+    const {
+        data: names,
+        error: isNamesErrored,
+        isLoading: isLoadingRegistrations,
+    } = useRegistrationNfts('name');
+    const {
+        data: auctionDetails,
+        error: isAuctionsErrored,
+        isLoading: isLoadingAuctions,
+    } = useGetUserAuctions();
+
+    const isLoadingCards = isLoadingAuctions || isLoadingRegistrations;
+
+    const router = useRouter();
+    const pathname = usePathname();
+    const account = useCurrentAccount();
+
+    const groupedAuctions = groupUserAuctions(auctionDetails, account?.address ?? '');
+
+    const filteredNames: (RegistrationNft | AuctionCard)[] = useMemo(() => {
+        const inAuction = groupedAuctions ?? [];
+        const owned = names ?? [];
+
+        switch (selectedFilter) {
+            case GroupedNamesFilter.All:
+                return [...owned, ...inAuction];
+            case GroupedNamesFilter.InAuction:
+                return inAuction;
+            case GroupedNamesFilter.Owned:
+                return owned;
+            default:
+                return owned;
+        }
+    }, [auctionDetails, names, selectedFilter]);
+
+    const noCardToDisplay =
+        !isLoadingCards && filteredNames.length === 0 && !isAuctionsErrored && !isNamesErrored;
+    const noAuctions = !auctionDetails || auctionDetails.length === 0;
+
+    function handleSubnameListClick(name: string): void {
+        router.push(MY_NAMES_ROUTE.path + `/${normalizeNameInput(name)}`);
+    }
 
     return (
-        <div className="flex flex-col w-full gap-y-lg items-center py-24 md:py-lg">
-            {updateNameDialog ? (
-                <UpdateNameDialog
-                    name={updateNameDialog}
-                    open
-                    setOpen={() => setUpdateNameDialog(null)}
-                />
-            ) : null}
-            {deleteNameDialog ? (
-                <DeleteNameDialog
-                    nft={deleteNameDialog}
-                    open
-                    setOpen={() => setDeleteNameDialog(null)}
-                />
-            ) : null}
-            <div className="pt-md">
-                <Title title="My names" testId="my-names-page" />
+        <>
+            <Breadcrumbs
+                items={[{ ...NAMES_CRUMB, isActive: pathname === NAMES_CRUMB.path }]}
+                trailingElement={<Button type={ButtonType.Outlined} text="Name" icon={<Add />} />}
+            />
+
+            <div className="flex">
+                <SegmentedButton>
+                    {Object.entries(GroupedNamesFilter).map(([key, value]) => (
+                        <ButtonSegment
+                            key={key}
+                            type={ButtonSegmentType.Rounded}
+                            label={value}
+                            selected={selectedFilter === value}
```

Comment:
can we disable the buttons if there’s nothing to show? something like:

```suggestion
                            selected={selectedFilter === value}
                             disabled={
     (value === GroupedNamesFilter.InAuction && noAuctions) ||
     (value === GroupedNamesFilter.Owned && !filteredNames.length)
 }
```